openmc/mgxs_library.py

```diff
@@ -457,102 +340,262 @@ def use_chi(self, use_chi):
         if npchi.shape != self.vector_shape:
             msg = 'Provided chi iterable does not have the expected shape.'
             raise ValueError(msg)
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
 
-        self._chi = npchi
+        i = self.temperatures.tolist().index(temperature)
+        self._chi[i] = npchi
 
         if self.use_chi is not None:
             self.use_chi = True
 
-    @scatter.setter
-    def scatter(self, scatter):
-        # Convert to a numpy array so we can easily get the shape for
-        # checking
+    def set_scatter_matrix(self, scatter, temperature=294.):
+        """This method sets the cross section for this XSdata object at the
+        provided temperature.
+
+        Parameters
+        ----------
+        scatter: np.ndarray
+            Scattering Matrix Cross Section
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset. Defaults
+            to 294K
+
+        See also
+        --------
+        openmc.mgxs_library.set_scatter_matrix_mgxs()
+
+        """
+        # Convert to a numpy array so we can easily get the shape for checking
         npscatter = np.asarray(scatter)
         check_iterable_type('scatter', npscatter, Real,
                             max_depth=len(npscatter.shape))
         check_value('scatter shape', npscatter.shape, [self.pn_matrix_shape])
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = self.temperatures.tolist().index(temperature)
+        self._scatter_matrix[i] = npscatter
 
-        self._scatter = npscatter
+    def set_multiplicity_matrix(self, multiplicity, temperature=294.):
+        """This method sets the cross section for this XSdata object at the
+        provided temperature.
+
+        Parameters
+        ----------
+        multiplicity: np.ndarray
+            Multiplicity Matrix Cross Section
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset. Defaults
+            to 294K
+
+        See also
+        --------
+        openmc.mgxs_library.set_multiplicity_matrix_mgxs()
 
-    @multiplicity.setter
-    def multiplicity(self, multiplicity):
-        # Convert to a numpy array so we can easily get the shape for
-        # checking
+        """
+        # Convert to a numpy array so we can easily get the shape for checking
         npmultiplicity = np.asarray(multiplicity)
         check_iterable_type('multiplicity', npmultiplicity, Real,
                             max_depth=len(npmultiplicity.shape))
         check_value('multiplicity shape', npmultiplicity.shape,
                     [self.matrix_shape])
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = self.temperatures.tolist().index(temperature)
```

Comment:
alternatively:

```
i = np.where(self.temperatures == temperature)
```

It certainly won't make a difference on performance (who cares about that for this anyway) since the number of temperatures will always be small, but just throwing it out there since we use that approach to compute array indices throughout the rest of the API.